drivers/counter/counter_nrfx_rtc.c

```diff
@@ -81,18 +94,97 @@ static u32_t counter_nrfx_read(struct device *dev)
 	return nrfx_rtc_counter_get(&get_nrfx_config(dev)->rtc);
 }
 
+/* Function calculates distance between to values assuming that one first
+ * argument is in front and that values wrap.
+ */
+static u32_t tick_diff(u32_t val, u32_t old, u32_t top)
+{
+	if (top & (top + 1)) {
+		/* if top is not 2^n-1 */
+		u32_t diff;
+
+		diff = (val > old) ? (val - old) : val + top + 1 - old;
+
+		return diff;
+	}
+
+	return (val - old) & top;
+}
+
+static int set_cc(struct device *dev, u8_t chan, u32_t val)
+{
+	const struct counter_nrfx_config *nrfx_config = get_nrfx_config(dev);
+	const nrfx_rtc_t *rtc = &nrfx_config->rtc;
+	NRF_RTC_Type  *reg = rtc->p_reg;
+	nrf_rtc_int_t int_mask;
+	nrf_rtc_event_t evt;
+	u32_t prev_val;
+	u32_t top;
+	u32_t now;
+
+	evt = (nrf_rtc_event_t)(offsetof(NRF_RTC_Type, EVENTS_COMPARE[chan]));
+	int_mask = (nrf_rtc_int_t)(NRF_RTC_INT_COMPARE0_MASK << chan);
+
+	nrf_rtc_event_clear(reg, evt);
+	top = counter_get_top_value(dev);
+	prev_val = nrf_rtc_cc_get(reg, chan);
+	nrf_rtc_cc_set(reg, chan, val);
+
+	/* If prev CC setting equals or is just in front of the counter then
+	 * there is a risk that before new CC will be set after enabling event
+	 * previous CC will generate COMPARE event. In such risk delay must be
```

Comment:
I'm not sure I understand the risk. I assume that it relates to the fact that the value written to the `CC` register will be actually applied in the peripheral at the next LFCLK tick. But isn't it the case also with the `EVTEN` register (I mean, isn't the application of its new value delayed in the same way)?

Reply:
I've just recalled that `EVTEN` usage here is useless (by @pabigot observation #15624). I've been trying to stress this procedure even more today since i've seen that it may be still possible to get interrupt on previous CC. I managed to caught it once. Working on improving test and algorithm.